pkg/operator/starter.go

```diff
@@ -175,6 +175,20 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 		controllerConfig.EventRecorder,
 	)
 
+	// This is the only controller that sets Upgradeable condition
+	cndController := NewConditionController(
+		"ConditionController",
+		operatorClient,
+		controllerConfig.EventRecorder,
+		[]operatorv1.OperatorCondition{
+			{
+				// The condition name should match the same condition in previous OCP release (4.11).
+				Type:   "CSISnapshotControllerUpgradeable",
+				Status: operatorv1.ConditionTrue,
+			},
+		},
```

Comment:
I have doubts about this controller. It seems strange to just force a condition without any validation, just for the sake of overriding a condition set in a previous release.

Would it make sense for this controller to reset conditions older than X hours instead?

Reply:
How do you know it's an old / stale condition? A controller should not update a condition if nothing changed in it, it can have quite old `LastTransitionTime` and be still correct. And I don't like resetting random conditions just because they have old `LastTransitionTime`.